### PR TITLE
[DRAFT] Outline of current work for Splunk Enterprise Receiver

### PR DESCRIPTION
**Pull request by MovieStoreGuy**
**Description:**
I had started the work on this some time ago but got lost in other efforts + sickness. Since others outside my company were interested in collaborating I wanted to share what I have so far.

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/12667

**Testing:**
Hadn't spent much on testing since I trying to find a design that was simple to extend.

**Documentation:** 
This is something I've struggled with on this receiver, trying to understand the expected config.

### CODE REVIEW

File: receiver/splunkenterprisereceiver/internal/endpoint/endpoint.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+// MarshalContent performs a HTTP call to the hostname + endpoint and will marshal the json into v.
+// Additional options can be passed in to allow modifying querying the endpoint.
+func (e Endpoint) MarshalContent(ctx context.Context, v any, opts ...func(eo *EndpointOption)) error {
+	eo := &EndpointOption{
+		Query: url.Values{
+			"output_mode": []string{"json"},
+		},
+		Method:     http.MethodGet,
+		Body:       http.NoBody,
+		Client:     http.DefaultClient,
```
Comment (atoulme) [inline receiver/splunkenterprisereceiver/internal/endpoint/endpoint.go:63]: Worth using the config.HttpClientSettings client here?

**Review by atoulme — COMMENTED** [review]
Nice start! Maybe we can come up with a metadata.yaml for the scraper and things will click from there.

**Comment by mtwo** [comment]
Note for readers: this is a proposal for retrieving health metrics from Splunk Enterprise for the purpose of monitoring it. I'd incorrectly thought that this was a HEC receiver (which would also be cool if it doesn't exist already!) until @MovieStoreGuy discussed it with me.

**Comment by atoulme** [comment]
This is now largely superseded by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21976. Would it make sense to close or rebase this PR?

**Comment by MovieStoreGuy** [comment]
Happy to close it

**Pull request closed without merging**